✨ Add Simulated Data Producer migration

jacobwoffenden opened the pull request:
Creates Kubernetes manifests for running migration against Cloud Platform RDS

I have validated that the Kubernetes manifests work, so merging this will run it again via GitHub Actions, duplicating the RDS data, but that is OK.

Commit:
Added Kubernetes manifests and actions workflow

Commit:
Add DB_NAME

Commit:
Updated main.py

murdo-moj committed:
Added adjudications data to postgres

murdo-moj commented on scripts/simulated-data-producer/configmap.yml:
```diff
@@ -0,0 +1,52 @@
+
+    engine = create_engine(f'postgresql://{DB_USERNAME}:{DB_PASSWORD}@{DB_ENDPOINT}:5432/{DB_NAME}')
+    df = pd.read_csv(os.path.join("data", adjudications_file))
+    df.to_sql('adjudications', engine)
```
I'm doing it through sqlalchemy to avoid having to write a schema for the table

PriyaBasker23 replied:
Can main,py and requirements.txt be as seperate file within scripts rather than yml file itself ?

jacobwoffenden replied:
Yes, with the following

```
kubectl --namespace ${NAMESPACE} create configmap \
  ${CONFIGMAP_NAME} \
  --from-file=main.py \
  --from-file=requirements.txt
```

However we'd have to replace it everytime, see https://stackoverflow.com/questions/38216278/update-k8s-configmap-or-secret-without-deleting-the-existing-one

murdo-moj committed:
Added psycopg2-binary to requirements

murdo-moj committed:
Added if_exists to postgres write

Commit:
Fixed file extension

Commit:
Merge branch 'feature/simulated-data-producer' of github.com:ministryofjustice/data-platform into feature/simulated-data-producer

murdo-moj committed:
Moved main.py to dedicated file rather than inline

murdo-moj committed:
Merge branch 'feature/simulated-data-producer' of https://github.com/ministryofjustice/data-platform into feature/simulated-data-producer

murdo-moj committed:
Put main.py back in configmap

murdo-moj committed:
Fix file pathing in configmap

Commit:
Add instructions for port forwarding to RDS

jacobwoffenden commented:
Update:

* Containers are not allowed to run as `root` on Cloud Platform, meaning we cannot use `docker.io/python:3.10-alpine3.17`

* I created an image based on Alpine 3.17, built using Apko (part of a wider piece of work to offer secure baseline images to Data Platform Products), however `pandas`, and by extension `numpy` do not install/build well on Alpine (see [link](https://stackoverflow.com/questions/49037742/why-does-it-take-ages-to-install-pandas-on-alpine-linux)). A _workaround_ for this is to use the Alpine packages for the required softwares
  * `requests==2.28.1` ➡️ https://pkgs.alpinelinux.org/package/v3.17/main/x86_64/py3-requests
  * `pandas==1.5.3` ➡️ https://pkgs.alpinelinux.org/package/v3.17/community/x86_64/py3-pandas
  * `SQLAlchemy==2.0.5.post1` ➡️ https://pkgs.alpinelinux.org/package/v3.17/community/x86_64/py3-sqlalchemy
  * `psycopg2-binary==2.9.5` ➡️ https://pkgs.alpinelinux.org/package/v3.17/community/x86_64/py3-psycopg2

Alternatively, I have just written up a document for connecting to RDS from our local machines, however I need to update access to the Cloud Platform namespace

murdo-moj committed:
Added more tables, dynamically create configmap

murdo-moj committed:
Merge branch 'feature/simulated-data-producer' of https://github.com/ministryofjustice/data-platform into feature/simulated-data-producer

murdo-moj committed:
Use alpine specific python dependencies

murdo-moj committed:
Reordered import for linting

murdo-moj committed:
Removed errant space

murdo-moj committed:
Satisfy isort

murdo-moj committed:
Tried to fix markdown linting

murdo-moj committed:
Further shorten lines

murdo-moj committed:
Removed testing variables

murdo-moj committed:
Added markdown lint file

murdo-moj committed:
Merge branch 'main' into feature/simulated-data-producer

Commit:
Add YAML start to markdown-lint
Updated configmap creation
Removed pip install from entrypoint.sh
Updated image in job.yml
Removed requirements.txt as they've been shifter to image

Commit:
Update main.py to use /tmp as /app is non writable due to being a volume
Update Job resources

github-actions[bot] committed:
Commit changes made by code formatters

Commit:
Update CODEOWNERS

Commit:
Adding backoffLimit so the job fails on first error

Commit:
Merge branch 'feature/simulated-data-producer' of github.com:ministryofjustice/data-platform into feature/simulated-data-producer
